src/compiler/scala/tools/nsc/typechecker/SyntheticMethods.scala

```diff
@@ -276,6 +276,11 @@ trait SyntheticMethods extends ast.TreeDSL {
       createMethod(nme.hashCode_, Nil, IntTpe) { m =>
         val accumulator = m.newVariable(newTermName("acc"), m.pos, SYNTHETIC) setInfo IntTpe
         val valdef      = ValDef(accumulator, Literal(Constant(0xcafebabe)))
+        val mixPrefix   =
+          Assign(
+            Ident(accumulator),
+            callStaticsMethod("mix")(Ident(accumulator),
+              Apply(gen.mkAttributedSelect(gen.mkAttributedSelect(mkThis, Product_productPrefix), Object_hashCode), Nil)))
```

Comment:
I don't understand why `Nil` is here, but maybe that means "use the default value; it wasn't supplied"?

Reply:
it's putting the `()` in `acc = this.productPrefix.hashCode()`